packages/common/src/ConfiguFile.ts

```diff
@@ -178,8 +216,54 @@ export class ConfiguFile {
     });
   }
 
-  private mergeSchemas(...schemas: ConfigSchema[]): ConfigSchema {
-    // Later schemas take precedence in case of key duplication.
-    return new ConfigSchema(_.merge({}, ...schemas.map((schema) => schema.keys)));
+  private static async registerModule(module: Record<string, unknown>) {
+    Object.entries(module).forEach(([key, value]) => {
+      // console.log('Registering:', key, value);
+
+      if (key === 'default') {
+        return;
+      }
+      if (typeof value === 'function' && 'type' in value) {
+        // console.log('Registering ConfigStore:', value.type);
+        ConfigStore.register(value as ConfigStoreConstructor);
+      } else if (typeof value === 'function') {
+        // console.log('Registering ConfigExpression:', key);
+        ConfigExpression.register(key, value);
+      } else {
+        // console.log('ignore registeree:', key);
+      }
+    });
+  }
+
+  static async registerModuleFile(filePath: string) {
+    const module = await importModule(filePath);
+    ConfiguFile.registerModule(module);
+  }
+
+  static async registerStore(type: string) {
+    const modulePath = await getConfiguHomeDir('cache', `/${type}.js`);
```

Comment:
Didn't we discuss changing `cache` to something else? Like `integrations` for example?